File: permissions/src/main/java/com/gabriel/permissions/domain/model/PermissionIDConverter.java
package com.gabriel.permissions.domain.model;

import com.gabriel.model.PermissionId;
import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;

@Converter(autoApply = true)
public class PermissionIDConverter implements AttributeConverter<PermissionId, String> {

    @Override
    public String convertToDatabaseColumn(PermissionId attribute) {
        if (attribute != null) {
            return attribute.getValue();
        }
        return null;
    }

    @Override
    public PermissionId convertToEntityAttribute(String dbData) {
        if (dbData != null) {
            return PermissionId.newBuilder().setValue(dbData).build();
        }
        return null;
    }
}

File: permissions/src/main/kotlin/com/gabriel/permissions/domain/model/PermissionIDConverter.kt
package com.gabriel.permissions.domain.model

import com.gabriel.model.PermissionId
import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter

@Converter(autoApply = true)
class PermissionIDConverter : AttributeConverter<PermissionId, String> {

    override fun convertToDatabaseColumn(attribute: PermissionId?): String? {
        return attribute?.value
    }

    override fun convertToEntityAttribute(dbData: String?): PermissionId? {
        return dbData?.let {
            PermissionId.newBuilder().setValue(it).build()
        }
    }
}
